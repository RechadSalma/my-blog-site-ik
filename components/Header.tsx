import Link from "next/link";
import { useEffect, useState } from "react";
import { Container, Nav, Navbar, Offcanvas } from "react-bootstrap";
// import Test from "bootstrap-icons/icons/linkedin.svg";

type Props = {};

export default function Header({}: Props) {
  const [closeCanvas, setCloseCanvas] = useState<boolean>(false);

  return (
    <Navbar bg="info" className="h-100 w-100 pt-0 pb-0">
      {/* {[false, "sm", "md", "lg", "xl", "xxl"].map((expand) => ( */}
      <Container fluid className="d-flex justify-content-center">
        <Navbar.Brand as="h1">
          <Link href="/">Ilshaad Blog</Link>
        </Navbar.Brand>
      </Container>
    </Navbar>
  );
}
