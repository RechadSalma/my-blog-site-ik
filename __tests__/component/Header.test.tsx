import { render, screen } from "@testing-library/react";
import Header from "../../components/Header";

describe("<Header/>", () => {
  it("renders the Ilshaad Blog heading", () => {
    const { container } = render(<Header />);

    const { getByRole, debug } = screen;

    // * the link (<a>) is under the heading but is a link & the heading of all pages
    const heading = getByRole("link", { name: /Ilshaad Blog/ });

    expect(heading).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
