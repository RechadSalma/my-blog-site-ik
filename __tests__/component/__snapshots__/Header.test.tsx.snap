// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders the Ilshaad Blog heading 1`] = `
<div>
  <nav
    class="h-100 w-100 pt-0 pb-0 navbar navbar-expand navbar-light bg-info"
  >
    <div
      class="d-flex justify-content-center container-fluid"
    >
      <h1
        class="navbar-brand"
      >
        <a
          href="/"
        >
          Ilshaad Blog
        </a>
      </h1>
    </div>
  </nav>
</div>
`;
